Integration example for external dns (#169)
lightstep/opentelemetry-examples: collector/external-dns/README.md

{: .callout}

## Requirements

* OpenTelemetry Collector Contrib v0.51.0+

## External DNS supports

ExternalDNS allows to configure external services, like [AWS Route53](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/aws.md), [Cloudflare DNS](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/cloudflare.md) and other ones. However for this example we provide configuration for CoreDNS.

## Prerequisites for local installation with CoreDNS

You must have a Lightstep Observability [access token](/docs/create-and-manage-access-tokens) for the project to report metrics to.

#### minikube

You can use any approach to managing your cluster, but the Makefile builds a cluster in `minikube`.

#### helm

We use helm charts to install some apps in this example.

## Running the Example

You can run this example with `make all` in this directory.

> After tests it's just need to call `make delete-cluster`.

Reviewer: ```suggestion
After tests just run `make delete-cluster`.
```

## Steps

### 1. Create a cluster

> First you'll need to create a cluster by a method of your choice. `minikube start` works well on Linux.

Reviewer: ```suggestion
First you'll need to create a cluster by a method of your choice. `minikube start` works well for local development on Linux.
```

### 2. Required libraries

> External DNS requires ETCD, DNS manager, for this example we picked CoreDNS. We add repos for coreDNS, otel-collector.

Reviewer: ```suggestion
External DNS requires etcd, DNS manager, for this example we picked CoreDNS. We add repos for coreDNS and otel-collector.
```

### 3. Installation

#### a. Install etcd

> etcd is used to manage etcd clusters, will be required for CoreDNS.

Reviewer: I don't think this should be "etcd clusters". I'm guessing this was meant to say "coreDNS clusters," but I leave it to you.

#### b. Install CoreDNS

> Installation CoreDNS should be pointed to etcd endpoint.

Reviewer: ```suggestion
Installation CoreDNS. It should be configured to point to the etcd endpoint.
```

... Something like that. I know this needs a little touch-up, but I'm not sure I got it right.

```sh
helm install my-coredns coredns/coredns -f values-coredns.yaml
```

#### c. Install ExternalDNS

> ExternalDNS configs also has to be pointed to etcd client.

Reviewer: ```suggestion
ExternalDNS also has to be configured to point to etcd client.
```

#### d. Install Nginx ingress

> For this example has been picked Nginx Ingress, but can be used any other one.

Reviewer: ```suggestion
For this example Nginx Ingress was chosen, but other ingress controllers may be used as well.
```
